src/api/java/com/minecolonies/api/crafting/ItemStorage.java

```diff
@@ -251,4 +251,25 @@ public int getRemainingDurablityValue()
     {
         return stack.getMaxDamage() - stack.getDamage();
     }
+
+    /**
+     * Is this an empty ItemStorage
+     * 
+     * @return true if empty
+     */
+    public boolean isEmpty()
+    {
+        return ItemStackUtils.isEmpty(stack) || amount <= 0;
+    }
+
+    /**
+     * Make a copy of the ItemStorage
+     * @return a copy
+     */
+    public ItemStorage copy()
```

Comment:
Could we add a .toImmutable too? should be useful

Reply:
I'll look at that in the next round of changes.